src/derivative_operators/abstract_operator_functions.jl

```diff
@@ -128,6 +128,13 @@ function LinearAlgebra.mul!(x_temp::AbstractVector{T}, A::Union{DerivativeOperat
     A.t[] += 1 # incrementing the internal time stamp
 end
 
+
+function LinearAlgebra.mul!(x_temp::AbstractVector{T}, A::InteriorOperator{T}, x::AbstractVector{T}) where T<:Real
+    convolve_interior!(x_temp, x, A)
```

Comment:
This also needs to handle the left/right because the boundary stencils have to do the non-centering when they are too large, so the edges needed to be handled directly as well. They should probably just be called CenteredDifferenceOperator (and DerivativeOperator should probably go away when these are ready).

Reply:
Wait a minute. Didn't we plan that we are separating the boundary and interior operations and therefore are using two different operators, `Q` and `L`?